drivers/fuel_gauge/sy24561/sy24561.c

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (c) 2025 Romain Paupe <rpaupe@free.fr>, Franck Duriez <franck.lucien.duriez@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Zephyr driver for SY24561 Battery Monitor
+ *
+ * Datasheet:
+ * https://www.silergy.com/download/downloadFile?id=4987&type=product&ftype=note
+ */
+#include <inttypes.h>
+#include <stdint.h>
+
+#include <zephyr/drivers/fuel_gauge.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "sy24561.h"
+
+#define DT_DRV_COMPAT silergy_sy24561
+
+LOG_MODULE_REGISTER(SY24561, CONFIG_FUEL_GAUGE_LOG_LEVEL);
+
+struct sy24561_config {
+	struct i2c_dt_spec i2c;
+};
+
+static int sy24561_read_reg(const struct device *dev, uint8_t reg, uint16_t *value)
+{
+	const struct sy24561_config *config = dev->config;
+	uint8_t buffer[2];
+	int const ret = i2c_write_read_dt(&config->i2c, &reg, sizeof(reg), buffer, sizeof(buffer));
+
+	if (ret != 0) {
+		LOG_ERR("i2c_write_read failed (reg 0x%02x): %d", reg, ret);
+		return ret;
+	}
+
+	*value = sys_get_be16(buffer);
+	LOG_DBG("reg[%02x]: %02x %02x => %04x", reg, buffer[0], buffer[1], *value);
+	return ret;
+}
+
+static int sy24561_write_reg(const struct device *dev, uint8_t reg, uint16_t value)
+{
+	const struct sy24561_config *config = dev->config;
+	uint8_t buffer[3] = {reg, 0, 0};
+
+	sys_put_be16(value, buffer + 1);
+
+	int const ret = i2c_write_dt(&config->i2c, buffer, sizeof(buffer));
+
+	if (ret != 0) {
+		LOG_ERR("i2c_write_read failed (reg 0x%02x): %d", reg, ret);
+		return ret;
+	}
+
+	return ret;
+}
+
+static int sy24561_get_voltage(const struct device *dev, int *voltage_uV)
+{
+	uint16_t voltage_reg = 0;
+
+	if (sy24561_read_reg(dev, SY24561_REG_VBAT, &voltage_reg) != 0) {
+		LOG_ERR("Impossible to get voltage register");
+		return -ENODEV;
```

Comment:
As mentioned in my last review: "You already set the return error value and message in `sy24561_read_reg()`". Rewrite error codes could hide the underlying error.
Please change to:
```suggestion
	int ret = sy24561_read_reg(dev, SY24561_REG_VBAT, &voltage_reg);
	if (ret)
	{
		return ret;
	}
```
and for all other occurrences.

Reply:
Sorry, I didn't see that some of your comments were folded by GitHub, so I missed them
Same for other comments later